spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/AbstractConfigProperties.java

```diff
@@ -98,70 +100,20 @@ public void setRetry(RetryProperties retry) {
 
 	/**
 	 * Kubernetes config retry properties.
+	 * @param initialInterval Initial retry interval in milliseconds.
+	 * @param multiplier Maximum interval for backoff.
+	 * @param maxInterval Maximum interval
+	 * @param maxAttempts Maximum number of attempts.
+	 * @param enabled Retry enabled or not
 	 */
-	public static class RetryProperties {
-
-		/**
-		 * Initial retry interval in milliseconds.
-		 */
-		private long initialInterval = 1000;
+	public record RetryProperties(@DefaultValue("1000") long initialInterval, @DefaultValue("1.1") double multiplier,
```

Comment:
this is the idea of the PR, move `RetryProperties` to a record, everything else are renames and tests